Lib/sqlite3/test/dbapi.py

```diff
@@ -933,6 +935,72 @@ def test_on_conflict_replace(self):
         self.assertEqual(self.cu.fetchall(), [('Very different data!', 'foo')])
 
 
+class MultiprocessTests(unittest.TestCase):
+    def setUp(self):
+        with sqlite.connect(TESTFN) as cx:
+            cx.execute("create table t(t)")
+
+    def tearDown(self):
+        unlink(TESTFN)
+
+    def test_rollback_if_db_is_locked(self):
+        # bpo-27334: ctx manager does not rollback if commit fails
+        CONNECTION_TIMEOUT = SHORT_TIMEOUT / 1000.  # Defaults to 30 ms
+        SCRIPT = f"""if 1:
+            import sqlite3
+            input()  # wait for parent process to start
+            cx = sqlite3.connect("{TESTFN}", timeout={CONNECTION_TIMEOUT})
```

Comment:
What is the purpose of this timeout?

Reply:
It's the SQLite [busy timeout](https://sqlite.org/c3ref/busy_timeout.html). In `sqlite3`, it defaults to 5 seconds. If we want this test to execute fast, we need a short timeout, since the test depends on SQL functions that times out :)